### PR TITLE
Add GDPR parameters in ad request

### PR DESCRIPTION
**Pull request by guillaume-sticky**
## Type of change

- [ ] Bugfix
- [ x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Add some parameters support for freewheel-ssp bidder adapter.
Especially GDPR parameters to be transmit to our ad server

- contact email of the adapter’s maintainer
gandouard@freewheel.tv

- [ *] official adapter submission

### CODE REVIEW

File: modules/freewheelSSPBidAdapter.js [hunk 1]
```diff
@@ -223,6 +223,20 @@ export const spec = {
       componentId: getComponentId(currentBidRequest.params.format)
     };
 
+    if (typeof currentBidRequest.params.gdpr !== 'undefined') {
```
Comment (harpere) [inline modules/freewheelSSPBidAdapter.js]: this wasn't done properly.  `buildRequests(bidRequests, bidderRequest)` has a second `bidderRequest` argument where you'll find the `gdpr` params.  please see the [doc](http://prebid.org/dev-docs/modules/consentManagement.html#buildrequests-integration).

Comment (harpere) [reply modules/freewheelSSPBidAdapter.js]: also, more specifically, I believe `gdpr_consent` could exist while `gdpr` is undefined.  So checking `gdpr !== 'undefined'` is not a good way to do this.

Comment (guillaume-sticky) [reply modules/freewheelSSPBidAdapter.js]: Thx, I'll have a look.

Comment (guillaume-sticky) [reply modules/freewheelSSPBidAdapter.js]: Changes have been made, thx again.

File: modules/freewheelSSPBidAdapter.js [hunk 2]
```diff
@@ -223,6 +223,24 @@ export const spec = {
...
+      requestParams._fw_gdpr_consent = bidderRequest.gdprConsent.consentString;
+
+      if (typeof currentBidRequest.gdprConsent.gdprApplies === 'boolean') {
+        requestParams._fw_gdpr = currentBidRequest.gdprConsent.gdprApplies;
+      }
+    }
+
+    var vastParams = currentBidRequest.params.vastUrlParams;
+    if (typeof vastParams === 'object') {
+      for (kye in vastParams) {
```
Comment (jsnellbaker) [inline modules/freewheelSSPBidAdapter.js]: Seems to be a small typo here; believe it should be `key`

File: modules/freewheelSSPBidAdapter.js [hunk 3]
```diff
@@ -223,6 +223,24 @@ export const spec = {
       componentId: getComponentId(currentBidRequest.params.format)
     };
 
+    // Add GDPR flag and consent string
+    if (bidderRequest.gdprConsent) {
+      requestParams._fw_gdpr_consent = bidderRequest.gdprConsent.consentString;
+
+      if (typeof currentBidRequest.gdprConsent.gdprApplies === 'boolean') {
+        requestParams._fw_gdpr = currentBidRequest.gdprConsent.gdprApplies;
```
Comment (jsnellbaker) [inline modules/freewheelSSPBidAdapter.js]: Should update `currentBidRequest` to `bidderRequest` in these two lines of code.

File: modules/freewheelSSPBidAdapter.js (hunk 2 above)
Comment (guillaume-sticky) [reply modules/freewheelSSPBidAdapter.js]: done, thx

File: modules/freewheelSSPBidAdapter.js (hunk 3 above)
Comment (guillaume-sticky) [reply modules/freewheelSSPBidAdapter.js]: done.

**Comment by mkendall07** [comment]
This pull request **introduces 1 alert** when merging 2a18f585270afe2b85a243b30f53ab9722c82d3d into 71503a77bc1235c73327c31d5cd8fd19beff0bb0 - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-57534af92165ab3ace606e40626fd20e09134075)

**new alerts:**

* 1 for Missing variable declaration

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: modules/freewheelSSPBidAdapter.js [hunk 4]
```diff
@@ -223,6 +223,24 @@ export const spec = {
...
+      requestParams._fw_gdpr_consent = bidderRequest.gdprConsent.consentString;
+
+      if (typeof bidderRequest.gdprConsent.gdprApplies === 'boolean') {
+        requestParams._fw_gdpr = bidderRequest.gdprConsent.gdprApplies;
+      }
+    }
+
+    var vastParams = currentBidRequest.params.vastUrlParams;
+    if (typeof vastParams === 'object') {
+      for (key in vastParams) {
```
Comment (jsnellbaker) [inline modules/freewheelSSPBidAdapter.js]: Update - I thought this was addressed in the commit, but it seems it wasn't.  The LGTM check noted this `key` variable isn't properly declared (missing a `let`).  Could you update this spot?

Comment (guillaume-sticky) [reply modules/freewheelSSPBidAdapter.js]: Done.

**Comment by jsnellbaker** [comment]
@guillaume-sticky Please submit a docs PR in the [docs repo](https://github.com/prebid/prebid.github.io) and make an update in your bidders page to add the following variable.
```
gdpr_supported: true
```

This can go directly below the variable for your 1.0 compliance.  This will allow your adapter to appear in a table that shows GDPR compliant adapters.  Thanks.

**Comment by guillaume-sticky** [comment]
will do. Thx!

**Pull request merged**